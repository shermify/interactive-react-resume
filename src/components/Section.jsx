import React from 'react';
import PropTypes from 'prop-types';

const Section = props => {
  const { id, title, lead, quoteBy, color, bgcolor } = props;
  return (
    <section id={id} style={{ backgroundColor: bgcolor, color }}>
      <div className="container section">
        <h2>{title}</h2>
        <p className="lead">
          {lead}
          <br />
          {quoteBy}
        </p>
        <hr />
        <div>{props.children}</div>
      </div>
    </section>
  );
};

Section.propTypes = {
  children: PropTypes.node,
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  lead: PropTypes.string.isRequired,
  quoteBy: PropTypes.string,
  color: PropTypes.string,
  bgcolor: PropTypes.string,
};

Section.defaultProps = {
  children: <div />,
  quoteBy: '',
  bgcolor: '#fff',
  color: '#333',
};

export default Section;
