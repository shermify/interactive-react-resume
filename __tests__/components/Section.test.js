import React from 'react';
import renderer from 'react-test-renderer';
import Section from '../../src/components/Section';

describe('<Section />', () => {
  const testProps = {
    id: 'Test',
    title: 'Section Test',
    lead: 'test lead',
    quoteBy: 'Daniel Sherman',
  };
  it('renders correctly', () => {
    const tree = renderer.create(
      <Section {...testProps}>Test text</Section>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
