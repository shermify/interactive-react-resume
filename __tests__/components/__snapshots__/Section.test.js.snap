// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> renders correctly 1`] = `
<section
  id="Test"
  style={
    Object {
      "backgroundColor": "#fff",
      "color": "#333",
    }
  }
>
  <div
    className="container section"
  >
    <h2>
      Section Test
    </h2>
    <p
      className="lead"
    >
      test lead
      <br />
      Daniel Sherman
    </p>
    <hr />
    <div>
      Test text
    </div>
  </div>
</section>
`;
